app/controllers/concerns/mfa_setup_concern.rb

```diff
@@ -1,6 +1,25 @@
 module MfaSetupConcern
   extend ActiveSupport::Concern
 
+  def user_next_authentication_setup_path!(final_path = nil)
+    case user_session[:selected_mfa_options]&.shift
```

Comment:
Related to @zachmargolis 's earlier comment, do we _need_ to be mutating the session like this? Or could we just peek at the first item?

```suggestion
  def user_next_authentication_setup_path(final_path = nil)
    case user_session[:selected_mfa_options].first
```

Reply:
I think we'd want to, so that refreshing a page or double submission doesn't lead to unexpected issues (unless the mutation is expected and handled, etc.)

Reply:
Yes.. I think we want to mutate it because we need to go through each mfa option. IF we only peak we will always just land users to their first selected MFA option. I could move the shifting of that array to the controller explicitly after we actually complete that setup, but a big portion of this is to properly redirect users to their next MFA setup, and also pop that out of the queue.

Reply:
Is it possible to use an index to track instead? It would allow for moving through the array while keeping the original data

Reply:
yea.